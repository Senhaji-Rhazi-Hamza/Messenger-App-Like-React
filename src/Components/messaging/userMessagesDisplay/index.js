import React from "react";
import UserHeaderSender from "./userHeaderSender";
import UserMessagesBox from "./userMessagesBox";
import UserSendBar from "./userSendBar";

export { default as UserHeaderSender } from "./userHeaderSender";
export { default as UserMessagesBox } from "./userMessagesBox";

const styles = {
  userMessageDisplayContainer: {
    height : '100%'
  },
};
const UserMessagesDisplay = ({ idUser }: { idUser: number }) => {
  return (
    <div style={{ ...styles.UserMessagesDisplayContainer }}>
      <div style={{ height: "20vh", width: "100%" }}>
        <UserHeaderSender idUser={idUser}> </UserHeaderSender>
      </div>
      <div style={{ overflow: "auto", height: "70vh",width: "100%" }}>
        <UserMessagesBox idUser={idUser}> </UserMessagesBox>
      </div>
      <div  style={{ height: "10vh",  width: "100%", marginTop : '10px'}}>
        <UserSendBar> </UserSendBar>
      </div>
    </div>
  );
};

export default UserMessagesDisplay;
