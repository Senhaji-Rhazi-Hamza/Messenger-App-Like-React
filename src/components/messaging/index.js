import React from "react";
import UserSearch from './userSearch'
import UserMessagesDisplay from './userMessagesDisplay'

export {default as UserSearch, UserElement, SearchBar, UsersList} from './userSearch'
export {default as UserMessagesDisplay,  UserHeaderSender} from './userMessagesDisplay'


const styles = {
  userMessagingUI: {
    display: "flex",
    flexDirection : 'row',
    height: '100%',
    minHeight: '100%',
  },
  userSearchContainer : {
      maxWidth : '35%',
      width : '35%',
      //position : 'fixed',
     // overflow : 'hidden'
      
   //   backgroundColor : 'red',
     // border: '2px dotted',
      //height : '100px'
  },
  userMessagingDisplayContainer : {
      maxWidth : '65%',
      width : '65%',
      
      height : '100%',
     
    //  minHeigh: '100vh'
  }
};
type Props = {
  idMainUser : number, // foo is required.
  idsUsers : Array<number>,
  idMainFriendUser? : number,
  updateIdMainFriendUser ? :(idMainFriendUser:nubmer) => void
};
const UserMessagingUI = ({idMainUser, idsUsers, idMainFriendUser, updateIdMainFriendUser}:Props) => {
  return (
      <div style = {{...styles.userMessagingUI}}> 
        {/* <div style> IM HERE</div> */}
       <div style = {{...styles.userSearchContainer}}>
        <UserSearch idMainUser = {idMainUser} idsUsers = {idsUsers}  idMainFriendUser = {idMainFriendUser} updateIdMainFriendUser = {updateIdMainFriendUser}> </ UserSearch>
       </div>
       <div style = {{...styles.userMessagingDisplayContainer}}>
        <UserMessagesDisplay idUser = {idMainUser}  idMainFriendUser = {idMainFriendUser}> </ UserMessagesDisplay>
       </div>
      </div>
    );
}

export default UserMessagingUI;
