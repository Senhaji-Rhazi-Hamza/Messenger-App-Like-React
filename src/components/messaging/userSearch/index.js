import React from "react";
import SearchBar from "./searchBar";
import UsersList from "./usersList";

export { default as UserElement } from "./userElement";
export { default as SearchBar } from "./searchBar";
export { default as UsersList } from "./usersList";

const styles = {
  userSearchContainer: {
    display: "flex",
    flexDirection : 'column',
  },
};
type Props = {
  idMainUser : number, // foo is required.
  idsUsers : Array<number>,
  updateIdMainFriendUser ? :(updateIdMainFriendUser:nubmer) => void,
  idMainFriendUser ? : number
};
const UserSearch = ({idMainUser, idsUsers, updateIdMainFriendUser, idMainFriendUser}:Props) => {
  return (
      <div style = {{...styles.userSearchContainer}}> 
      <div style = {{ height : '45%'}}> <SearchBar idUser = {idMainUser}> </SearchBar></div>
      <div style = {{ height : '75%'}}> <UsersList idsUsers = {idsUsers} updateIdMainFriendUser = {updateIdMainFriendUser}  idMainFriendUser = {idMainFriendUser} > </UsersList></div>
      </div>
    );
}

export default UserSearch;
