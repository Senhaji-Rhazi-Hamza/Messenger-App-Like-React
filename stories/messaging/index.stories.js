import React from 'react';

import  UserMessagingUI  from '../../src/components/messaging'

export default {
  title: 'UserMessagingUI',
  component: UserMessagingUI,
};
const idsUsers:Array<number> = [0,1,2,3,4,5,6,7]
const idMainUser:number = 0
export const Default = () => <UserMessagingUI idMainUser = {idMainUser} idsUsers = {idsUsers}></UserMessagingUI>;


