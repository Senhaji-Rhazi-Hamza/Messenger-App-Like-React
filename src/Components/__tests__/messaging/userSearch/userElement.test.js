import React from 'react';
import {UserElement} from '../../../messaging';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer
    .create(<UserElement idUser = {0}></UserElement>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});



// describe('UserElement', () => {
//   it('renders correctly', () => {
//     const wrapper = shallow(<UserElement />);
//     expect(wrapper).toMatchSnapshot();
//     // On the first run of this test, Jest will generate a snapshot file automatically.
//   });
// });