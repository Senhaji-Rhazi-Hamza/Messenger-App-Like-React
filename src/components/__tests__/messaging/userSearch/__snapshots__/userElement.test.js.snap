// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  class="hoverOnlyMe"
  onMouseOut={[Function]}
  onMouseOver={[Function]}
  style={
    Object {
      "border": "1px dotted gray",
      "borderRadius": "20px",
      "borderStyle": "ridge",
      "display": "flex",
      "flexDirection": "row",
      "height": "8hv",
      "margin": "1px 0",
      "maxHeight": "100px",
      "padding": 10,
      "with": "100%",
    }
  }
>
  <img
    alt="Avatar"
    src="https://images.pexels.com/photos/614810/pexels-photo-614810.jpeg?auto=compress&cs=tinysrgb&dpr=1&w=500"
    style={
      Object {
        "borderRadius": "50%",
        "float": "left",
        "height": "10%",
        "marginRight": "5%",
        "maxHeight": "100%",
        "maxWidth": "100%",
        "width": "10%",
      }
    }
  />
  <div
    style={
      Object {
        "display": "flex",
        "flexDirection": "column",
        "justifyContent": "space-between",
        "margin": "5px 0px 5px 0",
      }
    }
  >
    <div>
       User : 
      user_0
       
    </div>
    <div>
       Msg : 
      Message from user 1 to user 0
    </div>
  </div>
</div>
`;
